pkg/api/v1/types.go

```diff
@@ -1706,6 +1706,20 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+// DaemonEndpoint stores information about a single Daemon endpoint.
+type DaemonEndpoint struct {
+	// Port number of the given endpoint.
+	Port int `json:port`
+}
+
+// NodeDaemonEndpoints lists ports opened by daemons running on the Node.
+type NodeDaemonEndpoints struct {
+	// Endpoint on which Kubelet is listening.
+	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty"`
+	// Endpoint opened by Kubelet on which cAdvisor proxy is available.
+	CAdvisorProxyEndpoint DaemonEndpoint `json:"cAdvisorProxyEndpoint,omitempty"`
```

Comment:
AFAIK kube-ui might be using cadvisor directly.

On Wed, Sep 9, 2015 at 3:48 PM, Robert Bailey <notifications@github.com>
wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/12655#discussion_r39106657>
> :
>
> > @@ -1706,6 +1706,20 @@ type NodeSpec struct {
> >  	Unschedulable bool `json:"unschedulable,omitempty"`
> >  }
> >
> > +// DaemonEndpoint stores information about a single Daemon endpoint.
> > +type DaemonEndpoint struct {
> > +	// Port number of the given endpoint.
> > +	Port int `json:port`
> > +}
> > +
> > +// NodeDaemonEndpoints lists ports opened by daemons running on the Node.
> > +type NodeDaemonEndpoints struct {
> > +	// Endpoint on which Kubelet is listening.
> > +	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty"`
> > +	// Endpoint opened by Kubelet on which cAdvisor proxy is available.
> > +	CAdvisorProxyEndpoint DaemonEndpoint `json:"cAdvisorProxyEndpoint,omitempty"`
>
> @gmarek <https://github.com/gmarek> - Similar to the read-only kubelet
> port, I'm going to push back on adding the cadvisor proxy port into the v1
> API.
>
> Every port that we expose on the kubelet makes the work on securing the
> kubelet significantly more difficult. We need to configure authn/z
> handlers, set up SSL and potentially configure firewall rules to allow
> access.
>
> What consumers of the NodeDaemonEndpoints struct are going to use the
> cadvisor port? Do you know which system components hit cadvisor directly
> today?
>
> /cc @vishh <https://github.com/vishh>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12655/files#r39106657>.
>